### DISCUSSION TITLE
Use Java 17 as the future baseline?

### DISCUSSION THREAD

**Discussion by phax** [phax/ph-commons#37] [Polls]
Currently ph-commons and all my other ph* libraries use Java 11 as the baseline, as it provides all the needed features and has nice coverage. Unfortunately, Oracle ends the premium support for Java with end of September 2023 (see https://www.oracle.com/java/technologies/java-se-support-roadmap.html ). OpenJdk will support Java 11 until October 2024 (see e.g. https://access.redhat.com/articles/1299013). Popular frameworks like Spring Boot 3.x already require Java 17 as the baseline and other frameworks are following.

Updating to Java 11 as the baseline of all the libraries would mean:

* Java 17 or newer will be required for building the software
* Java 17 or newer will be required for running the software
* the Parent POM will enforce the usage of Java 17
* new major releases of all shared libraries as well as all related applications will be created
* Source code backwards compatibility to Java 11 will be dropped (this only effects very few places where APIs got Generic parameters in the runtime)

An update to Java 17 does not mean:

* That all the syntactic sugar of Java 17 will be used in the different components

Edit: the vote is open until October 31st, 2023

**Comment by QAnders** [comment]
As a little side-note both AWS and Azure is "pulling" (kind of) the support for Java 11 for various services. AWS Fargate for example, you can't set Java heap using Java 11, but you can using Java 17... Not sure about the underlaying functions there, but apparently something in the Java runtime making it difficult on Container (read Docker) services (which both AWS and Azure is using behind the scenes for Lambda/functions as well as container services (obviously)).

**Comment by phax** [comment]
As somebody is heavily voting for keeping the use of Java 11: can you please elaborate your reasons? Thanks

**Reply by beth-soptim** [reply]
My guess is that someone is using Phase 4 für BDEW market communication that requires Brainpool curves (which have been removed in JDK 17). With JDK 17+ one should use Bouncy Castle.

**Comment by phax** [comment]
This is the result per 31st of October 2023:

![grafik](https://github.com/phax/ph-commons/assets/196219/b845a4f2-b96f-4ef4-96e7-120a19e0de0b)

For me it is not umambiguous enough to make the switch at the moment - I will ask the same question again next year :)
